subprojects/packagefiles/avro-cpp/lang/c++/meson.build

```diff
@@ -0,0 +1,106 @@
+avro_flags = []
+avro_deps = []
+avro_public_deps = []
+
+fmt_dep = dependency(
+  'fmt',
+  required: true,
+)
+avro_deps += fmt_dep
+avro_public_deps += fmt_dep
```

Comment:
From a quick glance at the pkg-config template in avro, it also lists `@CODEC_PKG@` in the requires section, where CODEC_PKG is any of zlib, lzma and/or snappy. So two questions on that front:

1. Shouldn't the avro_public_deps include the compression dependencies?
2. Do we need to add lzma here as well? There exists a Meson wrapdb entry for it already

Reply:
> Shouldn't the avro_public_deps include the compression dependencies?

Do clients of the library need to directly link with the compression dependencies or use their headers?  Some of the headers in `lang/c++/include/avro` `#include` headers from `fmt`, but I don't see any that `#include` compressor headers.

Reply:
Yes, the codecs don't actually need to be linked, they are only used in this file: https://github.com/apache/avro/blob/557891c38c6c90eeaa43f1cb78ae8c3ae088009f/lang/c%2B%2B/impl/DataFile.cc